src/app/ipv/middleware.ts

```diff
@@ -207,48 +260,50 @@ function getCoiUpdateDetailsJourney(detailsToUpdate) {
   } else if (hasGivenNames) {
     return SUPPORTED_COMBO_EVENTS.UPDATE_GIVEN_NAMES;
   }
-}
+};
 
-function isInvalidDetailsToUpdate(detailsToUpdate) {
+function isInvalidDetailsToUpdate(detailsToUpdate: string[]): boolean {
   return (
     !detailsToUpdate ||
     (detailsToUpdate.includes("cancel") && detailsToUpdate.length > 1)
   );
 }
 
-function pageRequiresUserDetails(pageId) {
+export const pageRequiresUserDetails = (pageId: string): boolean => {
   return [
     PAGES.PAGE_IPV_REUSE,
-    PAGES.CONFIRM_NAME_DATE_BIRTH,
-    PAGES.CONFIRM_ADDRESS,
     PAGES.CONFIRM_DETAILS,
     PAGES.UPDATE_DETAILS,
-  ].includes(pageId);
-}
+  ].some((page) => page === pageId);
```

Comment:
This is a bit annoying! If you want to keep the existing method you'd have to cast the array to `string[]`, but this is ok too.

The real solution will be if/when TS supports proper bounded generics.